backends/tfhe-cuda-backend/src/bindings.rs

```diff
@@ -0,0 +1,1684 @@
+        num_radix_blocks: u32,
+        message_modulus: u32,
+        carry_modulus: u32,
+        pbs_type: PBS_TYPE,
+        lwe_per_glwe: u32,
+        storage_log_modulus: u32,
+        allocate_gpu_memory: bool,
+    );
+}
+extern "C" {
+    pub fn scratch_cuda_integer_decompress_radix_ciphertext_64(
+        streams: *const *mut ffi::c_void,
+        gpu_indexes: *const u32,
+        gpu_count: u32,
+        mem_ptr: *mut *mut i8,
+        encryption_glwe_dimension: u32,
+        encryption_polynomial_size: u32,
+        compression_glwe_dimension: u32,
+        compression_polynomial_size: u32,
+        lwe_dimension: u32,
+        pbs_level: u32,
+        pbs_base_log: u32,
+        num_radix_blocks: u32,
+        message_modulus: u32,
+        carry_modulus: u32,
+        pbs_type: PBS_TYPE,
+        storage_log_modulus: u32,
+        body_count: u32,
+        allocate_gpu_memory: bool,
+    );
+}
+extern "C" {
+    pub fn cuda_integer_compress_radix_ciphertext_64(
+        streams: *const *mut ffi::c_void,
+        gpu_indexes: *const u32,
+        gpu_count: u32,
+        glwe_array_out: *mut ffi::c_void,
+        lwe_array_in: *const ffi::c_void,
+        fp_ksk: *const *mut ffi::c_void,
+        num_nths: u32,
+        mem_ptr: *mut i8,
+    );
+}
+extern "C" {
+    pub fn cuda_integer_decompress_radix_ciphertext_64(
+        streams: *const *mut ffi::c_void,
+        gpu_indexes: *const u32,
+        gpu_count: u32,
+        lwe_array_out: *mut ffi::c_void,
+        glwe_in: *const ffi::c_void,
+        indexes_array: *const u32,
+        indexes_array_size: u32,
+        bsks: *const *mut ffi::c_void,
+        mem_ptr: *mut i8,
+    );
+}
+extern "C" {
+    pub fn cleanup_cuda_integer_compress_radix_ciphertext_64(
+        streams: *const *mut ffi::c_void,
+        gpu_indexes: *const u32,
+        gpu_count: u32,
+        mem_ptr_void: *mut *mut i8,
+    );
+}
+extern "C" {
+    pub fn cleanup_cuda_integer_decompress_radix_ciphertext_64(
+        streams: *const *mut ffi::c_void,
+        gpu_indexes: *const u32,
+        gpu_count: u32,
+        mem_ptr_void: *mut *mut i8,
+    );
+}
+pub const OUTPUT_CARRY_NONE: OUTPUT_CARRY = 0;
+pub const OUTPUT_CARRY_GENERATED: OUTPUT_CARRY = 1;
+pub const OUTPUT_CARRY_PROPAGATED: OUTPUT_CARRY = 2;
+pub type OUTPUT_CARRY = ffi::c_uint;
+pub const SHIFT_OR_ROTATE_TYPE_LEFT_SHIFT: SHIFT_OR_ROTATE_TYPE = 0;
+pub const SHIFT_OR_ROTATE_TYPE_RIGHT_SHIFT: SHIFT_OR_ROTATE_TYPE = 1;
+pub const SHIFT_OR_ROTATE_TYPE_LEFT_ROTATE: SHIFT_OR_ROTATE_TYPE = 2;
+pub const SHIFT_OR_ROTATE_TYPE_RIGHT_ROTATE: SHIFT_OR_ROTATE_TYPE = 3;
+pub type SHIFT_OR_ROTATE_TYPE = ffi::c_uint;
+pub const BITOP_TYPE_BITAND: BITOP_TYPE = 0;
+pub const BITOP_TYPE_BITOR: BITOP_TYPE = 1;
+pub const BITOP_TYPE_BITXOR: BITOP_TYPE = 2;
+pub const BITOP_TYPE_SCALAR_BITAND: BITOP_TYPE = 3;
+pub const BITOP_TYPE_SCALAR_BITOR: BITOP_TYPE = 4;
+pub const BITOP_TYPE_SCALAR_BITXOR: BITOP_TYPE = 5;
+pub type BITOP_TYPE = ffi::c_uint;
```

Comment:
it feels like all those could be improved on the rust side if you used enum class in the C++ code potentially, not for now but to be tested

Reply:
Yes totally, this is not something I'll try to do in this PR though.

Reply:
ok !